modules/cluster-logging-collector-rsyslog.adoc

```diff
@@ -0,0 +1,174 @@
+
+
+.Sample `sysconfig` ConfigMap based on the configuration
+
+[source,yaml]
+----
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: sysconfig.conf
+  namespace: openshift-logging
+  selfLink: /api/v1/namespaces/openshift-logging/configmaps/secure-forward
+  uid: 6fb09e64-0821-4939-b4bb-71b8100477dc
+  resourceVersion: '30996'
+  creationTimestamp: '2020-02-10T18:38:32Z'
+data:
+  sysconfig.conf: |
+    <store>
+     @type syslog_buffered
+     remote_syslog syslogserver.openshift-logging.svc.cluster.local
+     port 514
+     hostname fluentd-4nzfz
+     remove_tag_prefix tag
+     tag_key ident,systemd.u.SYSLOG_IDENTIFIER
+     facility local0
+     severity info
+     use_record true
+     payload_key message
+    </store> 
+----
+
+.Procedure
+
+To configure the Fluentd *syslog* plug-in:
+
+. Create a configuration file named `sysconfig.conf` containing the following
+parameters within the `<store>` stanza: 
+
+.. Specify the Fluentd *syslog* plug-in type, either `syslog` or `syslog_buffered`:
++
+----
+@type syslog_buffered <1>
+----
++
+<1> Specify the plug-in to use, either: `syslog` or `syslog_buffered`. 
+
+.. Configure the name, host, and port for your external syslog server:
++
+[source,yaml]
+----
+remote_syslog <fqdn or IP> <1>
+port <number> <2>
+hostname <name> <3>
+----
++
+<1> Specify the FQDN or IP address of the Rsyslog instance.
+<2> Specify the port of the Rsyslog instance.
+<3> Specify the name of the Rsyslog instance.
+
+If you specify two or more receivers, the Fluentd *syslog* plug-in uses these server nodes in a round-robin order.
++
+For example:
++
+[source,yaml]
+----
+remote_syslog rsyslogserver.openshift-logging.svc.cluster.local <2>
+port 514 <2>
+hostname fluentd-server <4>
+----
+
+.. Configure the other syslog varaibles as needed:
++
+----
+remove_tag_prefix <1>
+tag_key <key> <2>
+facility <value>  <3>
+severity <value>  <4>
+use_record <value> <5>
+payload_key message <6>
+----
++
+<1> Enter this value to remove the `tag` field from the syslog prefix.
+<2> Specify the field to set the syslog key.
+<3> Specify the syslog log facility or source. For values, see link:https://tools.ietf.org/html/rfc3164#section-4.1.1[RTF 3164].
+<4> Specify the syslog log severity. For values, see link:link:https://tools.ietf.org/html/rfc3164#section-4.1.1[RTF 3164].
+<5> Specify `true` to use the severity and facility from the record if available.
+<6> Use this field as the key to look on the record, to set the payload on the syslog message. Defaults to `message`.
++
+For example:
++
+[source,yaml]
+----
+facility local0
+severity info
+----
+
+. Create a ConfigMap named `sysconfig` in the `openshift-logging` namespace from the configuration file:
++
+----
+$ oc create configmap `sysconfig` --from-file=secure-forward.conf -n openshift-logging
+----
+
+. Refresh the `fluentd` Pod to apply the `sysconfig` ConfigMap:
++
+----
+$ oc delete pod --selector logging-infra=fluentd
```

Comment:
@mburke5678  Yes, we can apply the new configuration by 'oc delete pod --selector logging-infra=fluentd'.
@jcantrill  What is the better way to  redeploy the fluentd once there are changes in the configmap? Could we handle this in clo in future?

Reply:
> @mburke5678 Yes, we can apply the new configuration by 'oc delete pod --selector logging-infra=fluentd'.
> @jcantrill What is the best way to redeploy the fluentd once there are changes in the configmap?

The operator should redeploy fluent on changes or adding the configmaps but last option is to delete the pods which will force them to pick up the changes.